graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
 				}
 			}
 		case graph.Delete:
 			if !ignoreOpts.IgnoreMissing {
 				if _, exists := qs.indexOf(d.Quad); !exists {
-					return graph.ErrQuadNotExist
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadNotExist}
 				}
 			}
 		default:
-			return errors.New("memstore: invalid action")
+			return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrInvalidAction}
```

Comment:
They may be useful for the caller. At least he will not be forced to search for the right Delta by hand.